### PR TITLE
Update terminology for clarity and consistency

### PR DESCRIPTION
**Pull request by maxhutch**
# Citrine Python PR

## Description 
There are several changes here:
 * min/max -> lower/upper in ScalarRangeConstraint
 * CategoricalConstraint -> AcceptableCategoriesConstraint
 * max_size -> max_candidates in EnumeratedProcessor

Most of the change are made in a non-breaking way, with previous
patterns being deprecated. The hard changes are:
 * Categorical constraints will be deserialized as
   AcceptableCategoriesConstraint
 * ScalarRangeConstraint's min/max positions were flipped from
   max, min, min_inclusive, max_inclusive to min, max, min_inclusive,
   max_inclusive.  I suspect the old order was in error.

Also, the default value of max_candidates was changed from a very
large number to 1000.

Relates to: PLA-5934, PLA-5935, PLA-5936.

### PR Type:
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [x] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/informatics/constraints/scalar_range_constraint.py [hunk 1]
```diff
@@ -15,37 +16,106 @@ class ScalarRangeConstraint(Serializable['ScalarRangeConstraint'], Constraint):
...
-        self.max = max
-        self.min = min
-        self.min_inclusive = min_inclusive
-        self.max_inclusive = max_inclusive
+        self.lower_bound = lower_bound or min
+        self.upper_bound = upper_bound or max
+
+        # we have to be careful with None and boolean values
+        # None or False or True -> True, so that pattern doesn't work
+        self.lower_inclusive = True
```
Comment (bfolie) [inline src/citrine/informatics/constraints/scalar_range_constraint.py:76]: is this line necessary?

File: src/citrine/informatics/constraints/scalar_range_constraint.py [hunk 2]
```diff
@@ -15,37 +16,106 @@ class ScalarRangeConstraint(Serializable['ScalarRangeConstraint'], Constraint):
...
+        if lower_inclusive is not None:
+            self.lower_inclusive = lower_inclusive
+        elif min_inclusive is not None:
+            self.lower_inclusive = min_inclusive
+
+        self.upper_inclusive = True
```
Comment (bfolie) [inline src/citrine/informatics/constraints/scalar_range_constraint.py:82]: is this line necessary?

File: src/citrine/informatics/constraints/scalar_range_constraint.py (hunk 1 above)
Comment (maxhutch) [reply src/citrine/informatics/constraints/scalar_range_constraint.py:76]: Yes.  This sets the default if neither lower_inclusive nor min_inclusive are set.  We could also put this in an else statement.

File: src/citrine/informatics/constraints/scalar_range_constraint.py (hunk 2 above)
Comment (maxhutch) [reply src/citrine/informatics/constraints/scalar_range_constraint.py:82]: Yes.  This sets the default if neither upper_inclusive nor max_inclusive are set.  We could also put this in an else statement.

**Review by bfolie — APPROVED** [review]
Needs a version bump, but lgtm

**Review by bfolie — APPROVED** [review]

**Pull request merged**